### PR TITLE
Improve support for unexpected loss of network connection.

### PR DESCRIPTION
**Pull request by davivel**
Supplements https://github.com/owncloud/android/pull/1927

### CODE REVIEW

File: src/com/owncloud/android/lib/common/network/AdvancedSslSocketFactory.java [hunk 1]
```diff
@@ -223,19 +221,20 @@ public int hashCode() {
...
     public void setHostNameVerifier(X509HostnameVerifier hostnameVerifier) {
         mHostnameVerifier = hostnameVerifier;
     }
-    
+
     /**
-     * Verifies the identity of the server. 
-     * 
+     * Verifies the identity of the server.
+     * <p>
```
Comment (davigonz) [inline src/com/owncloud/android/lib/common/network/AdvancedSslSocketFactory.java]: Please delete this character

File: src/com/owncloud/android/lib/common/network/AdvancedSslSocketFactory.java [hunk 2]
```diff
@@ -324,22 +323,22 @@ private void verifyPeerIdentity(String host, int port, Socket socket) throws IOE
...
-	 * 
-	 * This is not cosmetic: not all the supported protocols are enabled by default. Too see an overview of 
-	 * supported and enabled protocols in the stock Security Provider in Android see the tables in
-	 * http://developer.android.com/reference/javax/net/ssl/SSLSocket.html.
-	 *  
-	 * @param socket
-	 */
+    /**
+     * Grants that all protocols supported by the Security Provider in mSslContext are enabled in socket.
+     * <p>
```
Comment (davigonz) [inline src/com/owncloud/android/lib/common/network/AdvancedSslSocketFactory.java]: Same here

File: src/com/owncloud/android/lib/resources/files/DownloadRemoteFileOperation.java [hunk 3]
```diff
@@ -163,15 +163,20 @@ private RemoteOperationResult downloadFile(OwnCloudClient client, File targetFil
...
 
-            } else if (status != FORBIDDEN_ERROR && status != SERVICE_UNAVAILABLE_ERROR){
+            } else if (status != FORBIDDEN_ERROR && status != SERVICE_UNAVAILABLE_ERROR) {
                 client.exhaustResponse(mGet.getResponseBodyAsStream());
 
             } // else, body read by RemoteOeprationResult constructor
 
             result = new RemoteOperationResult(isSuccess(status), mGet);
 
+        /*} catch (Exception e) {
```
Comment (davigonz) [inline src/com/owncloud/android/lib/resources/files/DownloadRemoteFileOperation.java]: Commented code here, it is needed?

File: src/com/owncloud/android/lib/common/network/WriteTimeoutEnforcer.java [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+ *
+ * @author David A. Velasco
+ */
+public class WriteTimeoutEnforcer {
+
+    private static final String TAG = WriteTimeoutEnforcer.class.getSimpleName();
+
+    private static final AtomicReference<WriteTimeoutEnforcer> mSingleInstance = new AtomicReference<>();
+
+    private static final String METHOD_NAME = "setSoWriteTimeout";
```
Comment (davigonz) [inline src/com/owncloud/android/lib/common/network/WriteTimeoutEnforcer.java]: I don't understand the name of this variable, why do you have included **So** on it, do you mean **Socket**?

Comment (davivel) [inline src/com/owncloud/android/lib/common/network/WriteTimeoutEnforcer.java]: That can't be changed. This code is searching for a "hidden" method called that way, that could or could not be available in the private packages of the framework.

See the link at the beginning of the class for the code that is being searched for.

Comment (davigonz) [inline src/com/owncloud/android/lib/common/network/WriteTimeoutEnforcer.java]: Sure

**Comment by davivel** [comment]
Was validated in https://github.com/owncloud/android/pull/1927 .

Let it go...

**Pull request merged**